File: clojure/exercism/word-count/src/word_count.clj
(ns word-count
  (:require [clojure.string :as str]))

(defn word-count [s]
  (let [remove-blank #(remove str/blank? %)]
    (-> s
        (str/replace #"[!&@$%^,:]" "")
        (str/lower-case)
        (str/split #" ")
        remove-blank
        frequencies)))

File: elixir/exercism/word-count/lib/word_count.ex
defmodule WordCount do
  @doc """
  Count the number of words in the sentence.

  Words are compared case-insensitively.
  """
  @spec count(String.t()) :: map
  def count(sentence) do
    String.downcase(sentence)
    |> String.replace(~r/[!&@$%^&:,]/, "")
    |> String.replace("_", " ")
    |> String.split
    |> Enum.frequencies
  end
end
